openedx/openedx-platform
# Expand mypy static type checking

## Description

The `edx-platform` codebase has `mypy` set up, but only a single django app (`learning_sequences`) is currently type checked.

This PR expands the type checking to include several django apps that I've developed:
* `openedx/core/djangoapps/content_staging`
* `openedx/core/djangoapps/content_libraries`
* `openedx/core/djangoapps/xblock`

The type checking found a couple actual bugs in the code, which I have now fixed. It also required a bunch of additional tweaks here are there to get the type checks passing.

## Testing instructions

[MyPy tests should be run by CI](https://github.com/openedx/edx-platform/blob/master/.github/workflows/ci-static-analysis.yml), so just confirm that they ran.

## Deadline

None

## Other information

Private ref MNG-3784
